collector/system_collector.go

```diff
@@ -0,0 +1,152 @@
+package collector
+
+import (
+	"nsxt_exporter/client"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/prometheus/client_golang/prometheus"
+	nsxt "github.com/vmware/go-vmware-nsxt"
+)
+
+func init() {
+	registerCollector("system", newSystemCollector)
+}
+
+type systemCollector struct {
+	systemClient client.SystemClient
+	logger       log.Logger
+
+	clusterStatus                    *prometheus.Desc
+	clusterNodeTotal                 *prometheus.Desc
+	clusterNodeUp                    *prometheus.Desc
+	clusterNodeStatus                *prometheus.Desc
+	applianceManagementServiceStatus *prometheus.Desc
+}
+
+func newSystemCollector(apiClient *nsxt.APIClient, logger log.Logger) prometheus.Collector {
+	nsxtClient := client.NewNSXTClient(apiClient, logger)
+	clusterStatus := prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "cluster", "status"),
+		"Status of NSX-T system controller and management cluster.",
+		nil,
+		nil,
+	)
+	clusterNodeTotal := prometheus.NewDesc(
```

Comment:
Do we still need total and number of Up nodes?
We already have individual UP/DOWN node, right?

Reply:
good catch, let me remove this, this is redundant